Assets/Plugin/thirdweb.jslib

```diff
@@ -1,13 +1,17 @@
 var plugin = {
-  ThirdwebInvoke: function (taskId, route, payload, cb) {
+  ThirdwebInvoke: function (taskId, route, payload, callback, cb) {
     // convert taskId from pointer to str and allocate it to keep in memory
     var id = UTF8ToString(taskId);
     var idSize = lengthBytesUTF8(id) + 1;
     var idPtr = _malloc(idSize);
     stringToUTF8(id, idPtr, idSize);
     // execute bridge call
     window.bridge
-      .invoke(UTF8ToString(route), UTF8ToString(payload))
+      .invoke(
+        UTF8ToString(route),
+        UTF8ToString(payload),
+        UTF8ToString(callback)
+      )
```

Comment:
in jslib land - currently `cb` is an `Action<...>` and that translates to JS as a function (...) -> void already. So you should be able to pass that down directly into the JS bridge.

I don't think we need to do the whole exposing the unity instance and use game object message passing for it.

the API would be so much cleaner and intuitive IMO

Reply:
cb is a one time callback which in this case contains nothing, whereas callback is the action passed into the listener so doesn't work here.
That being said I know where you're getting at with separate functions and passing an action like jsCallback to add to the listener, that works well but will need some tinkering to figure out what type of object is passed back into c#, in this case ContractEvent. Doable.